shared/js/intl_helper.js

```diff
@@ -42,6 +43,38 @@
         return false;
       }
     },
+    mozdatetime: {
+      create: function(options) {
+        const customOptions = Object.assign({}, options);
+        if (options.hour) {
+          customOptions.hour12 = navigator.mozHour12;
+        }
```

Comment:
so that if dev will do sth like:
```
var options = {
  ...
};
IntlHelper.define(name, type, options);
...
```
We don't modify the options provided.